File: src/LongestCommonPrefix/golang/src/solution/solution_test.go
package solution

import "testing"

func TestLongestCommonPrefix(t *testing.T) {
    tests := []struct {
        name  string
        strs  []string
        want  string
    }{
        {"All common prefix", []string{"flower", "flow", "flight"}, "fl"},
        {"No common prefix", []string{"dog", "racecar", "car"}, ""},
        {"Mixed length strings", []string{"interspecies", "interstellar", "interstate"}, "inters"},
        {"Single character common", []string{"circle", "circuit", "cirrus"}, "cir"},
        {"One string only", []string{"alone"}, "alone"},
        {"Empty string list", []string{}, ""},
    }

    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            if got := LongestCommonPrefix(tt.strs); got != tt.want {
                t.Errorf("LongestCommonPrefix() = %v, want %v", got, tt.want)
            }
        })
    }
}

File: src/LongestCommonPrefix/java/src/test/java/com/prefix/SolutionTest.java
package com.prefix;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

public class SolutionTest {
    @Test
    public void testLongestCommonPrefix() {
        Solution solution = new Solution();

        assertEquals("fl", solution.longestCommonPrefix(new String[] {"flower", "flow", "flight"}),
                     "Test failed with input: [\"flower\", \"flow\", \"flight\"]");

        assertEquals("", solution.longestCommonPrefix(new String[] {"dog", "racecar", "car"}),
                     "Test failed with input: [\"dog\", \"racecar\", \"car\"]");

        assertEquals("inters", solution.longestCommonPrefix(new String[] {"interspecies", "interstellar", "interstate"}),
                     "Test failed with input: [\"interspecies\", \"interstellar\", \"interstate\"]");
    }
}
